Improve SSASS and SLAM tooltips so they are more understandable

boubou19 opened the pull request:
|Before|After|
|:-:|:-:|
|![before_ssass](https://github.com/user-attachments/assets/bf2316e4-8263-4acb-9556-c35f0e2b5373)|![after_ssass](https://github.com/user-attachments/assets/f6e6cd0e-b2db-48bf-835f-13c4738326f9)|
|![before_slam](https://github.com/user-attachments/assets/79708688-b434-4e29-a1ae-9a90f8612e7d)|![after_slam](https://github.com/user-attachments/assets/b7f893c0-020b-4d86-a113-cad8af907474)|

Ruling-0 commented:
"produces x each cycle" instead of "produces each cycle x" sounds better for native English.

I personally think "cycles every second" sounds smoother than "each cycle lasts a second", especially as the introduction for what a cycle is. Another option might be "While running, completes a cycle every second".

Maybe mention that each hatch samples independently? If your goal is to clarify that there are 16 samples and not 1, since your new wording can still be read as a single sample.

boubou19 commented:
> Some quick feedback:
> 
> "produces x each cycle" instead of "produces each cycle x" sounds better for native English.
> 
> I personally think "cycles every second" sounds smoother than "each cycle lasts a second", especially as the introduction for what a cycle is. Another option might be "While running, completes a cycle every second".
> 
> Maybe mention that each hatch samples independently? If your goal is to clarify that there are 16 samples and not 1, since your new wording can still be read as a single sample.

I think I adressed your concerns, let me know if I missed some parts (don't mind the typo at the begining of "cycles every second", i corrected after i updated the screen but was too lazy to retake the screenshot)

Ruling-0 commented:
> > Some quick feedback:
> > "produces x each cycle" instead of "produces each cycle x" sounds better for native English.
> > I personally think "cycles every second" sounds smoother than "each cycle lasts a second", especially as the introduction for what a cycle is. Another option might be "While running, completes a cycle every second".
> > Maybe mention that each hatch samples independently? If your goal is to clarify that there are 16 samples and not 1, since your new wording can still be read as a single sample.
> 
> I think I adressed your concerns, let me know if I missed some parts (don't mind the typo at the begining of "cycles every second", I corrected after I updated the screen but was too lazy to retake the screenshot)

Yeah, addressed well, just caught another minor typo: "an a variance of 1" instead of "and a variance of 1"

boubou19 commented:
I got some time to edit the SLAM tooltip, so this PR is now ready for review and gather feedback.

srdr2k3 commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:
```diff
@@ -324,22 +324,72 @@ protected MultiblockTooltipBuilder createTooltip() {
-                    + "will waste energy!"
+                    + "will decrease efficiency!"
                     + EnumChatFormatting.GRAY)
+            .addInfo("Efficiency formula: Min(antimatter/matter, matter/antimatter))")
```
Definitely needs some colors here. Dark Blue for AM and IMO Purple for Matter, but it's up to you.
Also, unnecessary ")" at the end.
Please capitalize Matter and Antimatter too.

boubou19 replied:
I think i'll reuse Gold here for Matter to match the boost color

srdr2k3 commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:
```diff
@@ -324,22 +324,72 @@ protected MultiblockTooltipBuilder createTooltip() {
                     + EnumChatFormatting.GRAY)
+            .addInfo("Efficiency formula: Min(antimatter/matter, matter/antimatter))")
             .addInfo(
                 "Any EU that does not fit in laser hatches will be " + EnumChatFormatting.RED
```
"Any excess EU generated will be voided." is simpler and makes the point clear

boubou19 replied:
indeed, will change

srdr2k3 commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:
```diff
@@ -324,22 +324,72 @@ protected MultiblockTooltipBuilder createTooltip() {
+            .addInfo("3. Molten SC UMV Base: " + EnumChatFormatting.GOLD + "1.03")
+            .addInfo("4. Molten Black Dwarf Matter: " + EnumChatFormatting.GOLD + "1.04")
             .addSeparator()
             .addInfo("Enable wireless EU mode with screwdriver")
```
"Will transfer generated EU to your wireless network when right-clicked with a screwdriver."
Longer, but it should avoid confusion for the really clueless people out there.

srdr2k3 commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:
```diff
@@ -324,22 +324,72 @@ protected MultiblockTooltipBuilder createTooltip() {
+                    + ":")
+            .addInfo("1. Molten Copper: " + EnumChatFormatting.GOLD + "1.00")
+            .addInfo("2. Molten SC UIV Base: " + EnumChatFormatting.GOLD + "1.02")
+            .addInfo("3. Molten SC UMV Base: " + EnumChatFormatting.GOLD + "1.03")
```
SC and [Tier] need to be swapped to be consistent with the naming of superconductors. Honestly, don't shorten "Superconductor" but it's your call

boubou19 replied:
yeah I decided not to shorten the Superconductor here, because we have many free space on the line anyways, and because I don't like when there are too much accronyms.

srdr2k3 commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:
```diff
@@ -324,22 +324,72 @@ protected MultiblockTooltipBuilder createTooltip() {
+                    + EnumChatFormatting.GREEN
+                    + "ntimatter base energy"
+                    + EnumChatFormatting.GRAY
+                    + " value : "
```
Missed formatting here, or rather, it looks weird being one of the only unformatted words in the entire sentence.

boubou19 replied:
i updated the screenshots, let me know if it's better

srdr2k3 reviewed (commented):
Here are some nitpicks I could find.

boubou19 commented:
Ok so @srdr2k3 I tried to address your concerns, I also added the SSASS and the SLAM as machine types, for better NEI lookups. Let me know if things still need edits

Vlamonster commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterForge.java:
```diff
@@ -165,131 +165,169 @@ public IMetaTileEntity newMetaEntity(IGregTechTileEntity arg0) {
             .addInfo("Converts protomatter into antimatter")
             .addInfo(
-                "Consumes 10 000 000 + (" + EnumChatFormatting.DARK_AQUA
+                "Passively consumes 10 000 000 + (" + EnumChatFormatting.DARK_AQUA
```
`10,000,000`

boubou19 replied:
Looks like a missed `GTUtility.formatNumbers`

Ruling-0 commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:
```diff
@@ -312,35 +312,117 @@ public void onPreTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
+                    + EnumChatFormatting.UNDERLINE
+                    + "A"
+                    + EnumChatFormatting.RESET
+                    + "*("
```
Should probably be .GRAY after the reset, for consistency

boubou19 replied:
Fixed

boubou19 commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:
```diff
@@ -312,35 +312,117 @@ public void onPreTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
+                    + EnumChatFormatting.GOLD
+                    + "nergy production boost"
+                    + EnumChatFormatting.RESET
+                    + ":")
```
Fixed

boubou19 commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterForge.java:
```diff
@@ -165,131 +165,169 @@ public IMetaTileEntity newMetaEntity(IGregTechTileEntity arg0) {
+            .addInfo("Every cycle, the lowest amount of antimatter in the 16 antimatter hatches is recorded")
             .addInfo(
                 "All hatches with more than the lowest amount will " + EnumChatFormatting.RED
                     + "lose half the difference!"
```
I'm not sure I understand, as for me it's already done

Ruling-0 commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterForge.java:
```diff
@@ -165,131 +165,151 @@ public IMetaTileEntity newMetaEntity(IGregTechTileEntity arg0) {
+                "Each hatch will multiply the base production per hatch with a random number pulled from " + "N("
                     + baseSkew
+                    + " + "
+                    + EnumChatFormatting.AQUA
+                    + EnumChatFormatting.BOLD
+                    + EnumChatFormatting.UNDERLINE
+                    + "A"
+                    + EnumChatFormatting.RESET
                     + EnumChatFormatting.GRAY
-                    + ", 1) (Gaussian distribution with mean of "
-                    + baseSkew
-                    + ")")
-            .addInfo("The total change can be negative!")
+                    + ", 1)")
+            .addInfo("N(0.2, 1) refers to a normal distribution with a mean of 0.2 and a variance of 1")
```
It's looking very good - thanks for keeping the screenshots up to date. I had a quick idea for this line that reclaims a small amount of horizontal space, and also (in my opinion) flows a bit better.

```suggestion
            .addInfo("Each hatch will multiply the base production per hatch with a random number pulled from")
            .addInfo("N(" + baseskew + " + "
                    + EnumChatFormatting.AQUA
                    + EnumChatFormatting.BOLD
                    + EnumChatFormatting.UNDERLINE
                    + "A"
                    + EnumChatFormatting.RESET
                    + EnumChatFormatting.GRAY
                    + ", 1), which refers to a normal distribution with a mean of "+ baseskew + " + "
                    + EnumChatFormatting.AQUA
                    + EnumChatFormatting.BOLD
                    + EnumChatFormatting.UNDERLINE
                    + "A"
                    + EnumChatFormatting.RESET
                    + EnumChatFormatting.GRAY
                    + " and a variance of 1 ")
```

Alternatively, given N(M,S) syntax is now used only once, you could just remove it so it reads "pulled from a normal distribution with a mean of baseskew + A and a variance of 1".

boubou19 replied:
Yeah you are right, i think i'm going to yeet the normal distrubution notation, it does not bring any value to the tooltip

Ruling-0 reviewed (approved):
Looks good

Vlamonster commented on src/main/java/goodgenerator/blocks/tileEntity/AntimatterForge.java:
```diff
@@ -165,131 +165,150 @@ public IMetaTileEntity newMetaEntity(IGregTechTileEntity arg0) {
             .addSeparator()
-            .addInfo("Every cycle, the lowest amount of antimatter in the 16 antimatter hatches is recorded")
             .addInfo("Cycles every second")
+            .addInfo("Every cycle, the lowest amount of antimatter in the 16 antimatter hatches is recorded")
             .addInfo(
                 "All hatches with more than the lowest amount will " + EnumChatFormatting.RED
                     + "lose half the difference!"
                     + EnumChatFormatting.GRAY)
```
This might be clearer and concise:
"Each cycle, every antimatter hatch loses half the difference between its stored antimatter and the hatch with the smallest amount."

boubou19 replied:
I'm not convinced personnally, so you should ask around the discord to gather feedback, and do a followup PR if required
